amads/melody/contour/interpolation_contour.py

```diff
@@ -0,0 +1,272 @@
+
+        Returns
+        -------
+        float
+            Mean of the absolute gradient values
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.mean_gradient
+        30.0
+        """
+        return float(np.mean(np.abs(self.contour)))
+
+    @property
+    def gradient_std(self) -> float:
+        """Calculate the standard deviation of the interpolation contour gradients.
+
+        Returns
+        -------
+        float
+            Standard deviation of the gradient values
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.gradient_std
+        25.9272...
+        """
+        return float(np.std(self.contour))
+
+    @property
+    def direction_change_ratio(self) -> float:
+        """Calculate the ratio of direction changes in the interpolation contour.
+
+        This measures the number of changes in contour direction relative to the number
+        of interpolation lines (i.e. number of different gradient values).
+
+        Returns
+        -------
+        float
+            Ratio of direction changes to total gradient changes (between 0 and 1)
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3, 0.4], [60, 58, 64, 65, 63])
+        >>> ic.direction_change_ratio
+        2.0
+        """
+        # Get signs of all gradients
+        signs = np.sign(self.contour)
+
+        # Count direction changes (where consecutive signs are different)
+        direction_changes = np.sum(signs[:-1] != signs[1:])
+
+        # Count total gradient changes (where consecutive values are different)
+        total_changes = np.sum(self.contour[:-1] != self.contour[1:])
+
+        # Avoid division by zero
+        if total_changes == 0:
+            return 0.0
+
+        return float(direction_changes / total_changes)
+
+    @property
+    def class_label(self) -> str:
+        """Classify an interpolation contour into gradient categories.
+
+        The contour is sampled at 4 equally spaced points and each gradient is
+        normalized and classified into one of 5 categories:
+
+        - 'a': Strong downward (-2) - normalized gradient <= -1.45
+        - 'b': Downward (-1) - normalized gradient between -1.45 and -0.45
+        - 'c': Flat (0) - normalized gradient between -0.45 and 0.45
+        - 'd': Upward (1) - normalized gradient between 0.45 and 1.45
+        - 'e': Strong upward (2) - normalized gradient >= 1.45
+
+        Returns
+        -------
+        str
+            String of length 4 containing letters a-e representing the gradient
+            categories at 4 equally spaced points in the contour
+
+        Examples
+        --------
+        >>> ic = InterpolationContour([0, 0.1, 0.2, 0.3], [60, 58, 64, 65])
+        >>> ic.class_label
+        'aaee'
```

Comment:
Where did you get this example output from, are you sure it's right?

Reply:
Seems surprising that the final one would be strong upward

Reply:
updated with `method=` as discussed, with new, intuitive doctests